ui/app/ducks/swaps/swaps.js

```diff
@@ -536,3 +592,36 @@ export const signAndSendTransactions = (history, metaMetricsEvent) => {
     await forceUpdateMetamaskState(dispatch)
   }
 }
+
+export function fetchMetaSwapsGasPriceEstimates () {
+  return async (dispatch, getState) => {
+    const state = getState()
+    const priceEstimatesLastRetrieved = getSwapsPriceEstimatesLastRetrieved(state)
+    const timeLastRetrieved = priceEstimatesLastRetrieved || loadLocalStorageData('METASWAP_GAS_PRICE_ESTIMATES_LAST_RETRIEVED') || 0
+
+    dispatch(setSwapGasPriceEstimatesLoadingState(true))
+
+    let priceEstimates
+    if (Date.now() - timeLastRetrieved > 30000) {
+      priceEstimates = await fetchExternalMetaSwapGasPriceEstimates(dispatch)
+    } else {
+      const cachedPriceEstimates = loadLocalStorageData('METASWAP_GAS_PRICE_ESTIMATES')
+      priceEstimates = cachedPriceEstimates || await fetchExternalMetaSwapGasPriceEstimates(dispatch)
+    }
+
+    dispatch(setSwapGasPriceEstimates(priceEstimates))
+    dispatch(setSwapGasPriceEstimatesLoadingState(false))
+    return priceEstimates
+  }
+}
+
+async function fetchExternalMetaSwapGasPriceEstimates (dispatch) {
+  const priceEstimates = await fetchSwapsGasPrices()
```

Comment:
This function does not need to be exported and can even be easily worked into the one place where it is called

Reply:
4c8f00846